app/src/main/java/org/projectbuendia/client/sync/SyncAdapter.java

```diff
@@ -103,14 +69,23 @@
      * Select a phase by setting a boolean value of true for the appropriate key.
      */
     public enum SyncPhase {
-        SYNC_USERS,
-        SYNC_LOCATIONS,
-        SYNC_CHART_ITEMS,
-        SYNC_CONCEPTS,
-        SYNC_PATIENTS,
-        SYNC_OBSERVATIONS,
-        SYNC_ORDERS,
-        SYNC_FORMS
+        SYNC_USERS(R.string.syncing_users, new UsersSyncPhaseController()),
```

Comment:
To answer your question: each SyncPhase has a name returned by getName(), and the SyncAdapter has a Map from names to SyncPhase instances.  The caller requests a selection of phases by passing a bundle of flags, just as it does now, specifying the names of the phases.

In part my proposal is motivated by simplicity, and in part motivated by an inappropriate use of the name "Controller" -- these aren't controllers in the MVC sense, and the addition of "controller" to the name doesn't tell us any more information about what they are.

The interface seems to fit reasonably well -- why wouldn't incremental sync phases also have names and status messages?